app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,25 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.all
+    @posts = Post.where(author_id: params[:user_id])
     @user = User.find(params[:user_id])
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Post.find(params[:post_id])
     @user = User.find(params[:user_id])
-    @comments = Comment.all
+    @comments = @post.comments
+    @likes = @post.likes
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = Post.new(author: @user, title: params[:post][:title], text: params[:post][:text])
```

Comment:
- [ ] [optional] It would have been nice if you used private params, this will make your app more secure by controlling the model attributes the user can add. For more info click [here](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)

`NB this will apply to all your controllers except the application controller, kindly consider adding the private params.`